### ISSUE TITLE
Compiler errors

### ISSUE THREAD

**Issue by happyendings59** [MrMDavidson/SmsAndMmsMerger#1]
Does this code work yet?  i get these issues when compiling.

mcs Program.cs 
Program.cs(11,25): error CS1525: Unexpected symbol `;'
Program.cs(13,6): error CS1514: Unexpected symbol `public', expecting `.' or `{'
Program.cs(17,24): error CS1525: Unexpected symbol `!'
Program.cs(234,32): error CS1525: Unexpected symbol `.'
Program.cs(234,33): error CS1525: Unexpected symbol `.'
Program.cs(234,38): error CS1525: Unexpected symbol `]'
Program.cs(249,9): error CS1525: Unexpected symbol `using'
Program.cs(249,69): error CS1525: Unexpected symbol `)', expecting `;' or `}'
Program.cs(333,105): error CS1525: Unexpected symbol `!'
Program.cs(397,37): error CS1001: Unexpected symbol `.', expecting identifier
Program.cs(397,42): error CS1001: Unexpected symbol `]', expecting identifier
Program.cs(397,66): error CS0136: A local variable named `CarriageReturn' cannot be declared in this scope because it would give a different meaning to `CarriageReturn', which is already used in a `parent or current' scope to denote something else
Program.cs(397,79): error CS1001: Unexpected symbol `)', expecting identifier
Compilation failed: 13 error(s), 0 warnings

**Comment by MrMDavidson** [comment]
> mcs Program.cs

Am I right in guessing you're trying to compile this with Mono's compiler? That's not currently supported.

**Comment by happyendings59** [comment]
yes.  i'm just switching to ubuntu so a bit green.  is there one that does avail?

**Comment by MrMDavidson** [comment]
> yes. i'm just switching to ubuntu so a bit green. is there one that does avail?

Yup, you can find instructions over on [Microsoft's Ubuntu guide](https://learn.microsoft.com/en-au/dotnet/core/install/linux-ubuntu)

**Comment by happyendings59** [comment]
I was able to fix it with that guide.

the command will look something like this:

dotnet <path>SmsAndMmsMerger.dll -i file1.xml,file2.xml,file3.xml -o output.xml

it would be kind of useful if it would just process an entire folder of XML backups or had that easy option.

dotnet <path>/SmsAndMmsMerger-main/bin/Debug/net6.0/SmsAndMmsMerger.dll -i /<folder with XML file(s) path> *.xml -o output.xml

I had to do something like that where i just processed by "year in the filename" because of that other error I brought up with the odd XML tag and the special character issue and saved everything like output2022.xml, output2021.xml

I haven't merged the year output merges yet and tested it importing in my phone.

**Comment by MrMDavidson** [comment]
> it would be kind of useful if it would just process an entire folder of XML backups or had that easy option.

It wouldn't be too hard to implement _but_ it'd have to make assumptions about the "order" of the files for the merge if there's duplicate SMS/MMS content in any of the files.

In the interim you could probably script something that just repeatedly calls the merge tool with your preferred ordering.
